typing/env.ml

```diff
@@ -2846,6 +2853,36 @@ let add_exclave_lock env = add_lock Exclave_lock env
 
 let add_unboxed_lock env = add_lock Unboxed_lock env
 
+let add_local_env_lock env = add_lock Local_env_lock env
+
+let add_quotation_lock env =
+  add_lock Quotation_lock {env with stage = env.stage + 1}
+
+let add_splice_lock env = add_lock Splice_lock {env with stage = env.stage - 1}
```

Comment:
Similarly, this should be called `enter_splice`.

Moreover, having seen its only call site, I think this function should check that `stage > 0`, and raise error otherwise. (that means it should take a `~loc` parameter).

Reply:
What consistency are you refering to? I think `enter_quotation` and `enter_splice` is consistent naming?